Xamarin.Essentials/TextToSpeech/TextToSpeech.android.cs

```diff
@@ -0,0 +1,290 @@
+                        var localesavailable = Java.Util.Locale.GetAvailableLocales();
+                        foreach (var l in localesavailable)
+                        {
+                            try
+                            {
+                                var result = textToSpeech.IsLanguageAvailable(l);
+                                string name = null;
+
+                                switch (result)
+                                {
+                                    case LanguageAvailableResult.Available:
+                                        name = l.DisplayLanguage;
+                                        break;
+                                    case LanguageAvailableResult.CountryAvailable:
+                                        name = l.DisplayLanguage;
+                                        break;
+                                    case LanguageAvailableResult.CountryVarAvailable:
+                                        name = l.DisplayLanguage;
+                                        break;
+                                }
+
+                                locales.Add(new Locale(l.Language, l.Country, l.DisplayName));
+                            }
+                            catch (Exception ex)
+                            {
+                                Console.WriteLine("Error checking language; " + l + " " + ex);
+                            }
+                        }
+                    }
+                    catch (Exception ex)
+                    {
+                        var sb = new StringBuilder("Something went horribly wrong, defaulting to old implementation to get languages: ");
+                        sb.Append($"{ex.Message}");
+                        System.Diagnostics.Debug.WriteLine($"{sb.ToString()}");
+                        throw new InvalidProgramException($"{sb.ToString()}");
+                    }
+                }
+            }
+            else
+            {
+            }
+
+            return Task.FromResult(locales);
+        }
+
+        static Dictionary<string, string> iso3166Alpha2ToAlpha3Mapping;
+
+        public static Dictionary<string, string> ISO3166Alpha2ToAlpha3Mapping
+        {
+            get
+            {
+                if (iso3166Alpha2ToAlpha3Mapping == null)
+                {
+                    iso3166Alpha2ToAlpha3Mapping =
+                    CultureInfo
+                        .GetCultures(CultureTypes.SpecificCultures)
+                        .Select(region => new RegionInfo(region.LCID))
+                        .GroupBy(region => region.ThreeLetterISORegionName)
+                        .ToDictionary(group => group.Key, group => group.First().TwoLetterISORegionName);
+                }
+
+                return iso3166Alpha2ToAlpha3Mapping;
+            }
+        }
+    }
+
+    internal class TextToSpeechListener : Java.Lang.Object, global::Android.Speech.Tts.TextToSpeech.IOnInitListener, IDisposable
```

Comment:
this should have some sort of initialize method that can be awaited upon to ensure it is initialized.